### PR TITLE
[RW-7522][risk=no] CT workspace cards are disabled for non-CT users

### PR DESCRIPTION
**Pull request by aweng98**
https://user-images.githubusercontent.com/35533885/140594121-2a8f8ca2-4dad-4e5c-a5e7-f6b1872658a4.mov

### CODE REVIEW

File: ui/src/app/pages/workspace/workspace-list.tsx [hunk 1]
```diff
@@ -72,6 +80,11 @@ export class WorkspaceList extends React.Component<WithSpinnerOverlayProps, Stat
     }
   }
 
+  canAccessCTWorkspace(tierShortName: string) {
```
Comment (jmthibault79) [inline ui/src/app/pages/workspace/workspace-list.tsx:83]: there's a new function `hasTierAccess(profile: Profile, shortName)` in access-utils that we can use instead

File: ui/src/app/pages/workspace/workspace-card.tsx [hunk 2]
```diff
@@ -234,12 +240,14 @@ export const WorkspaceCard = fp.flow(withNavigation)(
             >
               <FlexColumn style={{marginBottom: 'auto'}}>
                 <FlexRow style={{ alignItems: 'flex-start' }}>
-                  <Clickable>
+                  <Clickable style={{cursor: disabled ? 'not-allowed' : 'pointer', ...styles}} disabled={disabled}>
+                    <TooltipTrigger content={'Controlled Tier workspace is not accessible.'} disabled={!disabled}>
                     <div style={styles.workspaceName}
                          data-test-id='workspace-card-name'
-                         onClick={() => this.onClick()}>
+                         onClick={() => !disabled ? this.onClick() : {}}>
```
Comment (jmthibault79) [inline ui/src/app/pages/workspace/workspace-card.tsx:247]: It shouldn't be necessary to check `disabled` explicitly.  Can we do something like moving the `onClick` to the `Clickable`?

File: ui/src/app/components/popups.tsx [hunk 3]
```diff
@@ -296,6 +296,7 @@ interface PopupTriggerProps {
   onOpen?: () => void;
   onClose?: () => void;
   side?: string;
+  disabled?: boolean;
```
Comment (jmthibault79) [inline ui/src/app/components/popups.tsx:299]: How does a disabled PopupTrigger differ from not rendering it at all?

**Comment by jmthibault79** [comment]
This seems to also cause the CT workspace to disappear from the homepage.  We should decide whether that is the desired behavior or not.

**Comment by jmthibault79** [comment]
Looking into the homepage issue... 

CT workspaces do not appear in the structure returned by the API because of `WorkspaceService.pruneInaccessibleRecentWorkspaces()` which checks `WorkspaceAuthService.enforceWorkspaceAccessLevel()` for **READER** or greater.  Users have the access level **NO_ACCESS** for Workspaces in a tier that they don't have access to, because they are not in the correct Terra Auth Domain.

**Review by jmthibault79 — APPROVED** [review]

**Review by NehaBroad — APPROVED** [review]

**Pull request merged**